crates/matrix-sdk-sqlite/src/state_store.rs

```diff
@@ -0,0 +1,1426 @@
+        // the error message: "cannot change into wal mode from within a transaction".
+        conn.execute_batch("PRAGMA journal_mode = wal;").await?;
+        conn.with_transaction(|txn| {
+            txn.execute_batch(include_str!("../migrations/state_store/001_init.sql"))
+        })
+        .await?;
+    }
+
+    conn.set_kv("version", vec![DATABASE_VERSION]).await?;
+
+    Ok(())
+}
+
+trait SqliteConnectionStateStoreExt {
+    fn set_kv_blob(&self, key: &[u8], value: &[u8]) -> rusqlite::Result<()>;
+
+    fn set_global_account_data(&self, event_type: &[u8], data: &[u8]) -> rusqlite::Result<()>;
+
+    fn set_room_account_data(
+        &self,
+        room_id: &[u8],
+        event_type: &[u8],
+        data: &[u8],
+    ) -> rusqlite::Result<()>;
+    fn remove_room_account_data(&self, room_id: &[u8]) -> rusqlite::Result<()>;
+
+    fn set_room_info(&self, room_id: &[u8], stripped: bool, data: &[u8]) -> rusqlite::Result<()>;
+    fn get_room_info(&self, room_id: &[u8], stripped: bool) -> rusqlite::Result<Option<Vec<u8>>>;
+    fn remove_room_info(&self, room_id: &[u8], stripped: Option<bool>) -> rusqlite::Result<()>;
+
+    fn set_state_event(
+        &self,
+        room_id: &[u8],
+        event_type: &[u8],
+        state_key: &[u8],
+        stripped: bool,
+        event_id: Option<&[u8]>,
+        data: &[u8],
+    ) -> rusqlite::Result<()>;
+    fn get_state_event_by_id(
+        &self,
+        room_id: &[u8],
+        event_id: &[u8],
+    ) -> rusqlite::Result<Option<Vec<u8>>>;
+    fn remove_room_state_events(
+        &self,
+        room_id: &[u8],
+        stripped: Option<bool>,
+    ) -> rusqlite::Result<()>;
+
+    fn set_member(
+        &self,
+        room_id: &[u8],
+        user_id: &[u8],
+        membership: &[u8],
+        stripped: bool,
+        data: &[u8],
+    ) -> rusqlite::Result<()>;
+    fn remove_room_members(&self, room_id: &[u8], stripped: Option<bool>) -> rusqlite::Result<()>;
+
+    fn set_profile(&self, room_id: &[u8], user_id: &[u8], data: &[u8]) -> rusqlite::Result<()>;
+    fn remove_room_profiles(&self, room_id: &[u8]) -> rusqlite::Result<()>;
+
+    fn set_receipt(
+        &self,
+        room_id: &[u8],
+        user_id: &[u8],
+        receipt_type: &[u8],
+        thread_id: &[u8],
+        event_id: &[u8],
+        data: &[u8],
+    ) -> rusqlite::Result<()>;
+    fn remove_room_receipts(&self, room_id: &[u8]) -> rusqlite::Result<()>;
+
+    fn set_display_name(&self, room_id: &[u8], name: &[u8], data: &[u8]) -> rusqlite::Result<()>;
+    fn remove_display_name(&self, room_id: &[u8], name: &[u8]) -> rusqlite::Result<()>;
+    fn remove_room_display_names(&self, room_id: &[u8]) -> rusqlite::Result<()>;
+}
+
+impl SqliteConnectionStateStoreExt for rusqlite::Connection {
+    fn set_kv_blob(&self, key: &[u8], value: &[u8]) -> rusqlite::Result<()> {
+        self.execute("INSERT OR REPLACE INTO kv_blob VALUES (?1, ?2)", (key, value))?;
```

Comment:
No need to number the parameters anymore with `INSERT OR REPLACE`.

```suggestion
        self.execute("INSERT OR REPLACE INTO kv_blob VALUES (?, ?)", (key, value))?;
```